src/Features/Core/Portable/CodeRefactorings/SyncNamespace/AbstractChangeNamespaceService.cs

```diff
@@ -141,13 +171,118 @@ await ChangeNamespaceToMatchFoldersAsync(solutionAfterNamespaceChange, id, decla
             }
         }
 
-        private ImmutableArray<ISymbol> GetDeclaredSymbolsInContainer(
-            SemanticModel semanticModel,
-            SyntaxNode node,
+        protected async Task<ImmutableArray<(DocumentId, SyntaxNode)>> TryGetApplicableContainersFromAllDocumentsAsync(
+            Solution solution,
+            ImmutableArray<DocumentId> ids,
+            TextSpan span,
+            CancellationToken cancellationToken)
+        {
+            // If the node specified by span doesn't meet the requirement to be an applicable container in any of the documents 
+            // (See `TryGetApplicableContainerFromSpanAsync`), or we are getting different namespace declarations among 
+            // those documents, then we know we can't make a proper code change. We will return null and the check 
+            // will return false. We use span of namespace declaration found in each document to decide if they are identical.            
+
+            var documents = ids.SelectAsArray(id => solution.GetDocument(id));
+            var containers = ArrayBuilder<(DocumentId, SyntaxNode)>.GetInstance(ids.Length);
+            var spanForContainers = PooledHashSet<TextSpan>.GetInstance();
+
+            try
+            {
+                foreach (var document in documents)
+                {
+                    var container = await TryGetApplicableContainerFromSpanAsync(document, span, cancellationToken).ConfigureAwait(false);
+                    containers.Add((document.Id, container));
+
+                    if (container is TNamespaceDeclarationSyntax)
+                    {
+                        spanForContainers.Add(container.Span);
+                    }
+                    else if (container is TCompilationUnitSyntax)
+                    {
+                        // In case there's no namespace declaration in the document, we used an empty span as key, 
+                        // since a valid namespace declaration node can't have zero length.
+                        spanForContainers.Add(default);
+                    }
+                    else
+                    {
+                        return default;
+                    }
+                }
+
+                return spanForContainers.Count == 1 ? containers.ToImmutable() : default;
```

Comment:
wait... you only return the container if you had one thing in it... so why is this method returning a list?  WHy is it not just returning the container if you found one, or null if you found 0/many?